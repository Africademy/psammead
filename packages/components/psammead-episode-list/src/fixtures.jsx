import React from 'react';
import VisuallyHiddenText from '@bbc/psammead-visually-hidden-text';

import EpisodeList from '.';

export const exampleEpisodes = [
  {
    id: '1',
    url: 'https://www.bbc.com',
    brandTitle: 'Magazine de la Culture',
    date: '4 Avril 2020',
    duration: 'Duration 59:00',
    time: '14:00',
  },
  {
    id: '2',
    url: 'https://www.bbc.com',
    brandTitle: 'Some other Brand',
    episodeTitle: 'Brandy Brand Talks About Hedgehogs',
    date: '4 Avril 2020',
    duration: 'Duration 59:00',
    time: '14:00',
  },
  {
    id: '3',
    url: 'https://www.bbc.com/blah',
    brandTitle: 'Some other Brand 2',
    episodeTitle: 'Brandy Brand Talks About Badgers',
    date: '4 Avril 2020',
    duration: 'Duration 59:00',
    time: '14:00',
  },
];

const rtlEpisode = {
  brandTitle: 'یونیورسٹی کی اندھیری',
  date: 'بی بی سی اردو ڈاٹ کام، کرا',
  duration: 'ایک گھنٹہ قبل',
  time: 'ریشان',
};

export const rtlEpisodes = [
  {
    id: '1',
    url: 'https://www.bbc.com',
    ...rtlEpisode,
  },
  {
    id: '2',
    url: 'https://www.bbc.com',
    ...rtlEpisode,
  },
  {
    id: '3',
    url: 'https://www.bbc.com/blah',
    ...rtlEpisode,
  },
];

export const renderEpisodes = (episodes, script, service, dir) => (
  <EpisodeList script={script} service={service} dir={dir}>
    {episodes.map(episode => (
      <EpisodeList.Episode
        key={episode.id}
        script={script}
        service={service}
        dir={dir}
        LinkElement={({ children, className }) => (
          <a href={episode.url} className={className}>
            {children}
          </a>
        )}
      >
        <VisuallyHiddenText>Audio</VisuallyHiddenText>
        <EpisodeList.BrandTitle>{episode.brandTitle}</EpisodeList.BrandTitle>
        <EpisodeList.EpisodeTitle>
          {episode.episodeTitle || `${episode.date}, ${episode.time}`}
        </EpisodeList.EpisodeTitle>
        <EpisodeList.MetaData>
          {episode.duration}
          {episode.episodeTitle && <span aria-hidden> | {episode.date}</span>}
        </EpisodeList.MetaData>
      </EpisodeList.Episode>
    ))}
  </EpisodeList>
);
