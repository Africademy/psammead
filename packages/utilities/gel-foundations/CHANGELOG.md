# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.0   | [PR#XX](https://github.com/BBC-News/psammead/pull/XX) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |
