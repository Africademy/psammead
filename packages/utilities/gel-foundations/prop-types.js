module.exports = require('./dist/prop-types'); // eslint-disable-line import/no-unresolved
