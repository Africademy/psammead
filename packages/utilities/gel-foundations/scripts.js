/* eslint-disable global-require */
module.exports = {
  news: require('./dist/scripts/news').default,
  arabic: require('./dist/scripts/arabic').default,
  burmese: require('./dist/scripts/burmese').default,
  cjkThai: require('./dist/scripts/cjk_thai').default,
  hindiNepaliBengali: require('./dist/scripts/hindi_nepali_bengali').default,
  latin: require('./dist/scripts/arabic').default,
  mundo: require('./dist/scripts/arabic').default,
  sinhala: require('./dist/scripts/sinhala').default,
  tamil: require('./dist/scripts/tamil').default,
};
