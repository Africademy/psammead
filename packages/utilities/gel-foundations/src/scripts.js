/* eslint-disable global-require */
module.exports = {
  latin: require('../dist/scripts/latin-and-cyrillic').default,
  cyrillic: require('../dist/scripts/latin-and-cyrillic').default,
  latinDiacritics: require('../dist/scripts/latin-with-diacritics').default,
  arabic: require('../dist/scripts/arabic').default,
  arabicPashto: require('../dist/scripts/arabic-pashto').default,
  bengali: require('../dist/scripts/bengali').default,
  burmese: require('../dist/scripts/burmese').default,
  hindi: require('../dist/scripts/devanagari-and-gurmukhi').default,
  nepali: require('../dist/scripts/devanagari-and-gurmukhi').default,
  sinhalese: require('../dist/scripts/sinhalese').default,
  tamil: require('../dist/scripts/tamil').default,
  thai: require('../dist/scripts/thai').default,
  chinese: require('../dist/scripts/no-ascenders-or-descenders').default,
  korean: require('../dist/scripts/no-ascenders-or-descenders').default,
  ethiopic: require('../dist/scripts/no-ascenders-or-descenders').default,
};
