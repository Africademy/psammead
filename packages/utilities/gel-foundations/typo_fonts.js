module.exports = require('./dist/typo_fonts'); // eslint-disable-line import/no-unresolved
