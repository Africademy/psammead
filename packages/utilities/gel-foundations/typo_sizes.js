module.exports = require('./dist/typo_sizes'); // eslint-disable-line import/no-unresolved
