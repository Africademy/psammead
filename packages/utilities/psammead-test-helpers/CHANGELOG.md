# Psammead Test Helpers Changelog

| Version | Description |
|---------|-------------|
| 0.1.2   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
