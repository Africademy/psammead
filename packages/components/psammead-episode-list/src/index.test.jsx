import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import * as scripts from '@bbc/gel-foundations/scripts';

import { renderEpisodes, exampleEpisodes, rtlEpisodes } from './fixtures';

describe('Episode List ', () => {
  shouldMatchSnapshot(
    'should render correctly with example episodes',
    renderEpisodes(exampleEpisodes, scripts.latin, 'news', 'ltr'),
  );
  shouldMatchSnapshot(
    'should render correctly with RTL episodes',
    renderEpisodes(rtlEpisodes, scripts.arabic, 'arabic', 'rtl'),
  );
  shouldMatchSnapshot(
    'should render correctly with a single episode',
    renderEpisodes([exampleEpisodes[0]], scripts.latin, 'news', 'ltr'),
  );
  shouldMatchSnapshot(
    'should render correctly with no episodes',
    renderEpisodes([], scripts.latin, 'news', 'ltr'),
  );
});
