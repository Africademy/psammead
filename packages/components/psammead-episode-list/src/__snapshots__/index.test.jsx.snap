// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode List  should render correctly with RTL episodes 1`] = `
.c7 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c2 {
  padding-top: 2px;
  display: inline-block;
  vertical-align: top;
}

.c2:dir(ltr) {
  padding-right: 16px;
}

.c2:dir(rtl) {
  padding-left: 16px;
}

.c3 {
  position: relative;
  width: 40px;
  height: 40px;
  border: solid 2px #000;
  border-radius: 50%;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  width: 0;
  height: 0;
  border-top: 7px solid transparent;
  border-bottom: 7px solid transparent;
  border-left: 12px solid #000;
  position: relative;
  left: 2px;
}

.c6 {
  display: inline-block;
}

.c1 {
  display: block;
}

.c1:hover span,
.c1:focus span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1:hover .play-button-wrapper,
.c1:focus .play-button-wrapper {
  background-color: #B80000;
  border-color: #B80000;
}

.c1:hover .play-icon,
.c1:focus .play-icon {
  border-left-color: white;
}

.c1:visited span {
  color: #6E6E73;
}

.c8 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #222222;
  display: block;
  font-weight: 700;
  font-size: 15px;
  line-height: 20px;
}

.c9 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #222222;
  display: block;
  font-size: 15px;
  line-height: 18px;
  margin: 4px 0;
}

.c10 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  display: block;
  font-size: 14px;
  line-height: 18px;
  color: #6E6E73;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
}

.c10 span {
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
}

.c0 {
  list-style: none;
  padding: 0;
  margin: 0;
}

.c0 li {
  padding: 16px 0;
}

.c0 li:first-child {
  padding-top: 0;
}

.c0 li:last-child {
  padding-bottom: 0;
}

.c0 li:not(:last-child) {
  border-bottom: 1px #BABABA solid;
}

@media (min-width:25rem) {
  .c2 {
    font-size: 8px;
  }
}

@media (min-width:25rem) {
  .c8 {
    font-size: 16px;
  }
}

@media (min-width:63rem) {
  .c9 {
    font-size: 14px;
  }
}

<ul
  class="c0"
>
  <li>
    <a
      class="c1"
      href="https://www.bbc.com"
    >
      <div>
        <div
          class="c2"
        >
          <div
            class="c3 play-button-wrapper"
          >
            <div
              class="c4"
            >
              <div
                class="c5 play-icon"
              />
            </div>
          </div>
        </div>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Audio
          </span>
          <span
            class="c8"
            dir="rtl"
          >
            یونیورسٹی کی اندھیری
          </span>
          <span
            class="c9"
            dir="rtl"
          >
            بی بی سی اردو ڈاٹ کام، کرا, ریشان
          </span>
          <span
            class="c10"
            dir="rtl"
          >
            ایک گھنٹہ قبل
          </span>
        </div>
      </div>
    </a>
  </li>
  <li>
    <a
      class="c1"
      href="https://www.bbc.com"
    >
      <div>
        <div
          class="c2"
        >
          <div
            class="c3 play-button-wrapper"
          >
            <div
              class="c4"
            >
              <div
                class="c5 play-icon"
              />
            </div>
          </div>
        </div>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Audio
          </span>
          <span
            class="c8"
            dir="rtl"
          >
            یونیورسٹی کی اندھیری
          </span>
          <span
            class="c9"
            dir="rtl"
          >
            بی بی سی اردو ڈاٹ کام، کرا, ریشان
          </span>
          <span
            class="c10"
            dir="rtl"
          >
            ایک گھنٹہ قبل
          </span>
        </div>
      </div>
    </a>
  </li>
  <li>
    <a
      class="c1"
      href="https://www.bbc.com/blah"
    >
      <div>
        <div
          class="c2"
        >
          <div
            class="c3 play-button-wrapper"
          >
            <div
              class="c4"
            >
              <div
                class="c5 play-icon"
              />
            </div>
          </div>
        </div>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Audio
          </span>
          <span
            class="c8"
            dir="rtl"
          >
            یونیورسٹی کی اندھیری
          </span>
          <span
            class="c9"
            dir="rtl"
          >
            بی بی سی اردو ڈاٹ کام، کرا, ریشان
          </span>
          <span
            class="c10"
            dir="rtl"
          >
            ایک گھنٹہ قبل
          </span>
        </div>
      </div>
    </a>
  </li>
</ul>
`;

exports[`Episode List  should render correctly with example episodes 1`] = `
.c7 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c2 {
  padding-top: 2px;
  display: inline-block;
  vertical-align: top;
}

.c2:dir(ltr) {
  padding-right: 16px;
}

.c2:dir(rtl) {
  padding-left: 16px;
}

.c3 {
  position: relative;
  width: 40px;
  height: 40px;
  border: solid 2px #000;
  border-radius: 50%;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  width: 0;
  height: 0;
  border-top: 7px solid transparent;
  border-bottom: 7px solid transparent;
  border-left: 12px solid #000;
  position: relative;
  left: 2px;
}

.c6 {
  display: inline-block;
}

.c1 {
  display: block;
}

.c1:hover span,
.c1:focus span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1:hover .play-button-wrapper,
.c1:focus .play-button-wrapper {
  background-color: #B80000;
  border-color: #B80000;
}

.c1:hover .play-icon,
.c1:focus .play-icon {
  border-left-color: white;
}

.c1:visited span {
  color: #6E6E73;
}

.c8 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #222222;
  display: block;
  font-weight: 700;
  font-size: 15px;
  line-height: 20px;
}

.c9 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #222222;
  display: block;
  font-size: 15px;
  line-height: 18px;
  margin: 4px 0;
}

.c10 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  display: block;
  font-size: 14px;
  line-height: 18px;
  color: #6E6E73;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
}

.c10 span {
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
}

.c0 {
  list-style: none;
  padding: 0;
  margin: 0;
}

.c0 li {
  padding: 16px 0;
}

.c0 li:first-child {
  padding-top: 0;
}

.c0 li:last-child {
  padding-bottom: 0;
}

.c0 li:not(:last-child) {
  border-bottom: 1px #BABABA solid;
}

@media (min-width:25rem) {
  .c2 {
    font-size: 8px;
  }
}

@media (min-width:25rem) {
  .c8 {
    font-size: 16px;
  }
}

@media (min-width:63rem) {
  .c9 {
    font-size: 14px;
  }
}

<ul
  class="c0"
>
  <li>
    <a
      class="c1"
      href="https://www.bbc.com"
    >
      <div>
        <div
          class="c2"
        >
          <div
            class="c3 play-button-wrapper"
          >
            <div
              class="c4"
            >
              <div
                class="c5 play-icon"
              />
            </div>
          </div>
        </div>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Audio
          </span>
          <span
            class="c8"
            dir="ltr"
          >
            Magazine de la Culture
          </span>
          <span
            class="c9"
            dir="ltr"
          >
            4 Avril 2020, 14:00
          </span>
          <span
            class="c10"
            dir="ltr"
          >
            Duration 59:00
          </span>
        </div>
      </div>
    </a>
  </li>
  <li>
    <a
      class="c1"
      href="https://www.bbc.com"
    >
      <div>
        <div
          class="c2"
        >
          <div
            class="c3 play-button-wrapper"
          >
            <div
              class="c4"
            >
              <div
                class="c5 play-icon"
              />
            </div>
          </div>
        </div>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Audio
          </span>
          <span
            class="c8"
            dir="ltr"
          >
            Some other Brand
          </span>
          <span
            class="c9"
            dir="ltr"
          >
            Brandy Brand Talks About Hedgehogs
          </span>
          <span
            class="c10"
            dir="ltr"
          >
            Duration 59:00
            <span
              aria-hidden="true"
            >
               | 
              4 Avril 2020
            </span>
          </span>
        </div>
      </div>
    </a>
  </li>
  <li>
    <a
      class="c1"
      href="https://www.bbc.com/blah"
    >
      <div>
        <div
          class="c2"
        >
          <div
            class="c3 play-button-wrapper"
          >
            <div
              class="c4"
            >
              <div
                class="c5 play-icon"
              />
            </div>
          </div>
        </div>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Audio
          </span>
          <span
            class="c8"
            dir="ltr"
          >
            Some other Brand 2
          </span>
          <span
            class="c9"
            dir="ltr"
          >
            Brandy Brand Talks About Badgers
          </span>
          <span
            class="c10"
            dir="ltr"
          >
            Duration 59:00
            <span
              aria-hidden="true"
            >
               | 
              4 Avril 2020
            </span>
          </span>
        </div>
      </div>
    </a>
  </li>
</ul>
`;

exports[`Episode List  should render correctly with no episodes 1`] = `<div />`;
